internal/settings.go

```diff
@@ -178,6 +185,18 @@ func (ds *DialSettings) Validate() error {
 	if ds.ImpersonationConfig != nil && len(ds.ImpersonationConfig.Scopes) == 0 && len(ds.Scopes) == 0 {
 		return errors.New("WithImpersonatedCredentials requires scopes being provided")
 	}
+	for _, opt := range ds.OpenTelemetryOpts {
+		switch o := opt.(type) {
+		case otelhttp.Option:
+			ds.OpenTelemetryOptsHTTP = append(ds.OpenTelemetryOptsHTTP, o)
+		case otelgrpc.Option:
+			ds.OpenTelemetryOptsGRPC = append(ds.OpenTelemetryOptsGRPC, o.(otelgrpc.Option))
```

Comment:
internal/settings.go:193:64: type assertion to the same type: o already has type otelgrpc.Option (S1040)